File: eMailManager2010/RibbonMailItem.vb
Imports Microsoft.Office.Tools.Ribbon

'Imports Outlook = Microsoft.Office.Interop.Outlook
Imports Office = Microsoft.Office.Core
Imports System.Diagnostics
Imports System.Windows.Forms

Imports Microsoft.Win32

Public Class RibbonMailItem
    Private LocRM As New Resources.ResourceManager("eMailManager.ResourceStrings", Me.GetType.Assembly)

    'Dim m_olMailItem As Outlook.MailItem

    Private Sub Ribbon1_Load(ByVal sender As System.Object, ByVal e As RibbonUIEventArgs) Handles MyBase.Load
        Me.Button1.Label = LocRM.GetString("MenuFileMessage")
    End Sub

    Private Sub Button1_Click(ByVal sender As System.Object, ByVal e As Microsoft.Office.Tools.Ribbon.RibbonControlEventArgs) Handles Button1.Click
        Try
            'If cSettings.MainForm = 0 Then
            'use default form
            Dim frm As New frmMain()
                With frm
                .m_olMailItem = CType(Globals.ThisAddIn.Application.ActiveInspector.CurrentItem, Outlook.MailItem)
                .bBatchFile = False
                .bSentMail = False

                    Globals.ThisAddIn.Application.ActiveInspector.Close(Outlook.OlInspectorClose.olDiscard)

                    .ShowDialog()
                    .Dispose()
                End With

            'ElseIf cSettings.MainForm = 1 Then
            '    ''use SCD form
            '    'Dim frm As New frmMain2()
            '    'With frm
            '    '    .m_olMailItem = Globals.ThisAddIn.Application.ActiveInspector.CurrentItem
            '    '    .bBatchFile = False
            '    '    .bSentMail = False

            '    '    Globals.ThisAddIn.Application.ActiveInspector.Close(Outlook.OlInspectorClose.olDiscard)

            '    '    .ShowDialog()
            '    '    .Dispose()
            '    'End With
            'End If

        Catch ex As System.Exception

        End Try
    End Sub

End Class

File: eMailManager/RibbonMailItem.cs
using Microsoft.Office.Tools.Ribbon;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Outlook = Microsoft.Office.Interop.Outlook;

namespace eMailManager
{
    public partial class RibbonMailItem
    {
        private void RibbonMailItem_Load(object sender, RibbonUIEventArgs e)
        {
            this.buttonFileMessage.Label = GlobalConfig.localResourceManager.GetString("MenuFileMessage");
        }

        private void buttonFileMessage_Click(object sender, RibbonControlEventArgs e)
        {
            var mailItems = new List<Outlook.MailItem>();
            mailItems.Add((Outlook.MailItem)Globals.ThisAddIn.Application.ActiveInspector().CurrentItem);

            Globals.ThisAddIn.Application.ActiveInspector().Close(Outlook.OlInspectorClose.olDiscard);

            var frm = new FormMain(mailItems, false);
            frm.ShowDialog();
        }
    }
}
